### PR TITLE
Added information about maintenance mode

### PR DESCRIPTION
**Pull request by meker12**
- Description of maintenance mode
- Link to enable and disable and configure maintenance mode for Magento Commerce
- Link to information about how Magento Commerce Cloud uses maintenance mode on deploy phase

## This PR is a:

- [x] Content update

whatsnew
Added information about Magento Maintenance mode to the [About Magento modes article](https://devdocs.magento.com/guides/v2.2/config-guide/bootstrap/magento-modes.html) in the Magento configuration guide.

### CODE REVIEW

**Comment by jeff-matthews** [comment]
Fixes #2950

File: guides/v2.1/config-guide/bootstrap/magento-modes.md [hunk 1]
```diff
@@ -93,6 +100,15 @@ In production mode:
...
 -   Errors are logged to the file system and are never displayed to the user.
 
+## Maintenance mode
+
+Run the Magento software in maintenance mode when you need to take your site offline, for example when you are performing installation and upgrade tasks. When your site is in maintenance mode, normal site visitors get a `Service Temporarily Unavailable` message when they navigate to your store. You can configure maintenance mode to allow visitors from authorized IP addresses to view the store normally.
+
+You can [enable and disable maintenance mode]({{ page.baseurl }}/install-gde/install/cli/install-cli-subcommands-maint.html) and configure exempt IP addresses using Magento CLI commands.
+
+For customers using {{site.data.var.ece}}, maintenance mode is enabled automatically for the duration of the deploy phase. When the deployment completes sucessfully, maintenance mode is turned off, and the Magento application is put back into production mode. See [Deployment hooks]({{ page.baseurl }}/cloud/reference/discover-deploy.html).
```
Comment (hguthrie) [inline guides/v2.1/config-guide/bootstrap/magento-modes.md]: Passive. Previously, the word "run" is used to describe the mode activation. So, in keeping with this spirit, maybe say
"...Magento runs in maintenance mode during the deploy phase."

Also, "turned on" and "put back" sounds like you're explaining an old tube television. Maybe.. "When the deployment completes successfully, Magento returns to running in normal production mode."

Also, Magento application sounds better, but wonder if we should use a variable.

Also, successfully is misspelled.

File: guides/v2.2/config-guide/bootstrap/magento-modes.md [hunk 2]
```diff
@@ -66,10 +75,7 @@ For more information, see [Set the Magento mode]({{ page.baseurl }}/config-guide
 
 ## Developer mode
 
-You should run the Magento software in developer mode when you're extending or customizing it.
-
-{:.bs-callout .bs-callout-info}
-[{{site.data.var.ece}}]({{ page.baseurl }}/cloud/bk-cloud.html) supports production mode only.
+You can run the Magento software in developer mode when you're extending or customizing it.
```
Comment (hguthrie) [inline guides/v2.2/config-guide/bootstrap/magento-modes.md]: contractions do not translate nice

File: guides/v2.2/config-guide/bootstrap/magento-modes.md [hunk 3]
```diff
@@ -47,9 +44,21 @@ You can run Magento in any of the following *modes*:
...
+		</tr>
 </tbody>
 </table>
 
+{:.bs-callout .bs-callout-info}
+[{{site.data.var.ece}}]({{ page.baseurl }}/cloud/bk-cloud.html) supports only the production and maintenance modes.
+
 ## Default mode
 
 As its name implies, default mode is how the Magento software operates if no other mode is specified. Default mode enables you to deploy the Magento application on a single server without changing any settings. However, default mode is not as optimized for production as is production mode.
```
Comment (hguthrie) [inline guides/v2.2/config-guide/bootstrap/magento-modes.md]: This comment just shows the use of different references to Magento. "Magento software" hmmm. I'm sure there is an inconsistency across docs about this. Application? Software? Product? ugh. @jeff-matthews What say you?

Comment (jeff-matthews) [reply guides/v2.2/config-guide/bootstrap/magento-modes.md]: I think of Magento as a platform or framework, so "Magento software" is awkward for me. I wouldn't refer to "Node.js software" or "Ruby on Rails software"; I would simply refer to it as Node, Ruby on Rails, or Magento. This is a good topic for team conversation about style.

**Comment by jeff-matthews** [comment]
@meker12, it's best to wait for reviewer approval prior to merging.

File: guides/v2.1/config-guide/bootstrap/magento-modes.md [hunk 4]
```diff
@@ -93,6 +99,15 @@ In production mode:
...
 -   Errors are logged to the file system and are never displayed to the user.
 
+## Maintenance mode
+
+Run Magento in maintenance mode to take your site offline while you complete maintainenance, upgrade, or configuration tasks.  In maintenance mode, the site redirects visitors to a default `Service Temporarily Unavailable` page.
+
+You can create a [custom mainenance page]({{ page.baseurl }}/comp-mgr/trouble/cman/maint-mode.html#compman-trouble-maint-create.html), manually enable and disable maintenance mode, and configure maintenance mode to allow visitors from authorized IP addresses to view the store normally. See [enable and disable maintenance mode]({{ page.baseurl }}/install-gde/install/cli/install-cli-subcommands-maint.html).
+
+If you are using {{site.data.var.ece}}, Magento applicationruns in maintenance mode during the deploy phase. When the deployment completes successfully, Magento returns to running in production mode. See [Deployment hooks]({{ page.baseurl }}/cloud/reference/discover-deploy.html).
```
Comment (hguthrie) [inline guides/v2.1/config-guide/bootstrap/magento-modes.md]: There needs to be a space after "application"

File: guides/v2.1/install-gde/install/cli/install-cli-subcommands-maint.md [hunk 5]
```diff
@@ -10,6 +10,16 @@ functional_areas:
   - Setup
 ---
 
+
+Magento uses [maintenance mode]({{ page.baseurl }}/config-guide/bootstrap/magento-modes.html#maintenance-mode) to disable bootstrapping; for example, while you're maintaining, upgrading, or reconfiguring your site.
```
Comment (hguthrie) [inline guides/v2.1/install-gde/install/cli/install-cli-subcommands-maint.md]: contraction

File: guides/v2.2/config-guide/bootstrap/magento-modes.md [hunk 6]
```diff
@@ -94,6 +99,14 @@ In production mode:
 -   You can enable and disable cache types only using the [command line]({{ page.baseurl }}/config-guide/cli/config-cli-subcommands-cache.html#config-cli-subcommands-cache-en).
 
 	You _cannot_ enable or disable cache types using the Magento Admin
+	
+## Maintenance mode
+
+Run Magento in maintenance mode to take your site offline while you complete maintainenance, upgrade, or configuration tasks.  In maintenance mode, the site redirects visitors to a default `Service Temporarily Unavailable` page.
+
+You can create a [custom mainenance page]({{ page.baseurl }}/comp-mgr/trouble/cman/maint-mode.html#compman-trouble-maint-create.html), manually enable and disable maintenance mode, and configure maintenance mode to allow visitors from authorized IP addresses to view the store normally. See [enable and disable maintenance mode]({{ page.baseurl }}/install-gde/install/cli/install-cli-subcommands-maint.html).
```
Comment (hguthrie) [inline guides/v2.2/config-guide/bootstrap/magento-modes.md]: maintenance is misspelled

File: guides/v2.1/config-guide/bootstrap/magento-modes.md [hunk 7]
```diff
@@ -93,6 +99,15 @@ In production mode:
 -   Static view files are not materialized and URLs for them are composed on the fly. Static view files are served from the {% glossarytooltip 0bc9c8bc-de1a-4a06-9c99-a89a29c30645 %}cache{% endglossarytooltip %} only.)
 -   Errors are logged to the file system and are never displayed to the user.
 
+## Maintenance mode
+
+Run Magento in maintenance mode to take your site offline while you complete maintainenance, upgrade, or configuration tasks.  In maintenance mode, the site redirects visitors to a default `Service Temporarily Unavailable` page.
+
+You can create a [custom mainenance page]({{ page.baseurl }}/comp-mgr/trouble/cman/maint-mode.html#compman-trouble-maint-create.html), manually enable and disable maintenance mode, and configure maintenance mode to allow visitors from authorized IP addresses to view the store normally. See [enable and disable maintenance mode]({{ page.baseurl }}/install-gde/install/cli/install-cli-subcommands-maint.html).
```
Comment (hguthrie) [inline guides/v2.1/config-guide/bootstrap/magento-modes.md]: maintenance is misspelled

File: guides/v2.1/config-guide/bootstrap/magento-modes.md [hunk 8]
```diff
@@ -93,6 +99,15 @@ In production mode:
 -   Static view files are not materialized and URLs for them are composed on the fly. Static view files are served from the {% glossarytooltip 0bc9c8bc-de1a-4a06-9c99-a89a29c30645 %}cache{% endglossarytooltip %} only.)
 -   Errors are logged to the file system and are never displayed to the user.
 
+## Maintenance mode
+
+Run Magento in maintenance mode to take your site offline while you complete maintainenance, upgrade, or configuration tasks.  In maintenance mode, the site redirects visitors to a default `Service Temporarily Unavailable` page.
```
Comment (hguthrie) [inline guides/v2.1/config-guide/bootstrap/magento-modes.md]: maintenance is misspelled

**Pull request merged**